pkg/utils/webhook/certificates.go

```diff
@@ -26,9 +27,60 @@ const (
 	certSecretName = "landscaper-webhook-cert"
 )
 
+// The DNSNamesConfig holds the list of DNS names used in the certificates.CertificateSecretConfig.
+type DNSNamesConfig interface {
+	Get() []string
+}
+
+// The DNSNamespacedNameConfig derives the DNS names from the service name and service namespace.
+type DNSNamespacedNameConfig struct {
+	config []string
+}
+
+// The DNSWebhookURLConfig derives the DNS names directly from the webhook url.
+type DNSWebhookURLConfig struct {
+	config []string
+}
+
+func (c *DNSNamespacedNameConfig) Set(name, namespace string) {
+	c.config = []string{
+		name,
+		fmt.Sprintf("%s.%s", name, namespace),
+		fmt.Sprintf("%s.%s.svc", name, namespace),
+	}
+}
+
+func (c *DNSNamespacedNameConfig) Get() []string {
+	return c.config
+}
+
+func (c *DNSWebhookURLConfig) Set(webhookURL string) error {
+	parsedURL, err := url.Parse(webhookURL)
+	if err != nil {
+		return err
+	}
+	dnsName, _, err := net.SplitHostPort(parsedURL.Host)
+	if err != nil {
+		// If SplitHostPort fails here, it is due to a missing port.
+		// In this case the host of the parsed URL is used directly.
+		c.config = []string{
+			parsedURL.Host,
+		}
+	} else {
+		c.config = []string{
+			dnsName,
+		}
+	}
+	return nil
+}
+
+func (c *DNSWebhookURLConfig) Get() []string {
+	return c.config
+}
+
 // GenerateCertificates generates the certificates that are required for a webhook. It returns the ca bundle, and it
 // stores the server certificate and key locally on the file system.
-func GenerateCertificates(ctx context.Context, kubeClient client.Client, certDir, namespace, name string) ([]byte, error) {
+func GenerateCertificates(ctx context.Context, kubeClient client.Client, certDir, namespace string, dnsNamesConfig DNSNamesConfig) ([]byte, error) {
```

Comment:
what is the benefit of the interface `DNSNamesConfig` instead of just passing a string array directly?

something like
```suggestion
func GenerateCertificates(ctx context.Context, kubeClient client.Client, certDir, namespace string, dnsNamesConfig []string) ([]byte, error) {
```

Reply:
You are right, actually not much. I have removed the interface and pass the list directly.